apps/04TowersOfHanoi.js

```diff
@@ -16,23 +26,85 @@ function printStacks() {
     console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+
+
+// Instructions for Spec 1
+// below is our function declaration, or function statement
+// function statements should contain:
+//
+//
+// 1. the name of the function  (we already have this)
+// 2. arguments in parentheses seperated by commas (we are given these, startStack and endStack)
+// 3. javascript statements that define the function enclosed in curly brackets(we are told to startStack.push() and endStack.pull())
+//
+//
+// After we define our function we then need to call this function from another function towersOfHanoi()
+// End Spec 1
+
+
+function movePiece(startStack, endStack) {
     // Your code here
 
+    // startStack.pop();
+    // stacks[startStack].pop()
+    var stackPiece = stacks[startStack].pop();
+    stacks[endStack].push(stackPiece);
+
 }
 
-function isLegal() {
-    // Your code here
 
+// Instructions for Spec 2
+// 1. the function isLegal() takes two arguments, startStack and endStack
+// 2. It should check to see if the block(stackPiece) being moved,
+// from startStack is smaller (less than operator) than the last the block in endStack
+// 3. return true if it is allowed
+// 4. return false if it is not
+// 5. if endStack has no block any block can be moved there (if endStack is empty the move is legal)
+//
+//
+//
+
+// what is a legal move?
+// if something exists in start stack and nothing exits in endStack then the
+
+function isLegal(startStack, endStack) {
+    // Your code here
+    var startStackArray = stacks[startStack];
+    var endStackArray   = stacks[endStack];
+
+    if (startStackArray.length === 0) {
+      return false;
```

Comment:
This is good! If i am trying to move from b to c which are both empty, this would stop me from breaking the program